app/models/SpecialMentionGuarantee.scala

```diff
@@ -75,13 +68,13 @@ final case class SpecialMentionGuarantee(additionalInfo: String) extends Special
       Right(None)
     }
     else if(amountString.length > 18) {
-      Left(AmountStringTooLong("Amount is longer than 18 characters"))
-    }
-    else if(amountString.matches("^[0-9]*\\.[0-9]{2}$")) {
-      Right(Some(BigDecimal(amountString)))
+      Left(AmountStringTooLong("Guarantee amount should be not be longer than 18 characters"))
     }
     else {
-      Left(AmountStringInvalid(s"Invalid String for amount: $amountString $cutIndex $additionalInfo $guaranteeReference $currencyOpt"))
-    }
+      Try(BigDecimal(amountString)) match {
+        case Success(bigDecimal) => Right(Some(bigDecimal))
+        case Failure(_) =>  Left(AmountStringInvalid("Amount String Invalid"))
```

Comment:
The error message could be changed to "The specified amount is not a valid representation of a decimal." (to reflect the NumberFormatException being thrown)